website/content/posts/gating-image-promotion-on-gitlab.md

```diff
@@ -0,0 +1,382 @@
+supply chain security best practices will be prevented from consuming the image. Those poor users...
+
+To mitigate these issues, we introduce a gating step that validates the image, like consumers would,
+*before* promoting the image. As part of this validation, let's define a minimal Enterprise Contract
+policy configuration file, called `policy.yaml`, that captures the validity requirements:
+
+```yaml
+---
+identity:
+  issuer: https://gitlab.com
+  subject: https://gitlab.com/lucarval/sign-attest-poc//.gitlab-ci.yml@refs/heads/main
+```
+
+Whoa where did those values come from!?
+
+This is part of the [identity-based] signature workflow implemented by Sigstore.
+
+The `issuer` refers to the entity that is responsible for *issuing* the identity. In our case, this
+is `https://gitlab.com` because we are running our pipeline from a git repository hosted on
+[gitlab.com](https://gitlab.com).
+
+The `subject` is the identity GitLab decided to assign to this particular pipeline. It is derived
+from the git repository, the file that defines the pipeline, and the git reference. If we were to
+run this same pipeline on a different git repository, or just on a different branch, the `subject`
+would contain a different value.
+
+By specifying these values in our `policy.yaml`, we are stating that we expect our images to always
+come from this particular branch in this particular repository.
+
+```yaml
+# [...]
+
+validate:
+  stage: promote
+  image:
+    name: quay.io/enterprise-contract/ec-cli:snapshot
+    entrypoint: [""]
+  variables:
+    DOCKER_IMAGE_NAME: $CI_REGISTRY_IMAGE:$CI_COMMIT_SHORT_SHA
+    GIT_REVISION: $CI_COMMIT_SHA
+  script:
+    - ec validate image --image "${DOCKER_IMAGE_NAME}" --policy policy.yaml --output yaml --show-successes
+
+tag-latest:
+  needs: [validate]  # <- added
+  stage: promote
+  # [...]
+```
+
+Now we have the guarantee that when the image is promoted, it already has the expected signature and
+SLSA Provenance attestation. Cool!
+
+## Advanced Validation
+
+Our `policy.yaml` is very minimal. The Enterprise Contract validation is ensuring the image is
+signed with the expected identity, and that it also contains a SLSA Provenance also signed with the
+same identity. This is great, but we can, and will, take it further!
+
+Let's use one of the community policy rules to ensure the SLSA Provenance correctly captures the git
+information. The [slsa_source_correlated] policy package is meant for exactly this use case.
+
+To use it, let's update our `policy.yaml`:
+
+```yaml
+---
+identity:
+  subject: https://gitlab.com/lucarval/sign-attest-poc//.gitlab-ci.yml@refs/heads/main
+  issuer: https://gitlab.com
+sources:
+  - policy:
+      - github.com/enterprise-contract/ec-policies//policy/lib
+      - github.com/enterprise-contract/ec-policies//policy/release
+    config:
+      include:
+        - slsa_source_correlated
```

Comment:
I think there's a missing sentence or two to explain the updates to the policy.yaml and what they're for. Maybe also a link to the slsa_source_correlated docs?

Reply:
+1

Reply:
Ack. It's linked to the docs on the paragraph above that explains why we're making the change. I rephrased the last sentence to highlight that we are adding the `sources` section, and added a little comment on the yaml to make it obvious (like in the other examples)